monitor/subscription.go

```diff
@@ -247,53 +265,74 @@ func (s *Subscription) AddNodes(nodes ...string) error {
 
 // AddNodeIDs adds nodes
 func (s *Subscription) AddNodeIDs(nodes ...*ua.NodeID) error {
+
+	requests := make([]Request, len(nodes))
+
+	for _, node := range nodes {
+		requests = append(requests, Request{
+			NodeID:         node,
+			MonitoringMode: ua.MonitoringModeReporting,
+		})
+	}
+	_, err := s.AddMonitorItems(requests...)
+	return err
+}
+
+// AddMonitorItems adds nodes with monitoring parameters to the subscription
+func (s *Subscription) AddMonitorItems(nodes ...Request) ([]Item, error) {
 	s.mu.Lock()
 	defer s.mu.Unlock()
 
 	if len(nodes) == 0 {
 		// some server implementionations allow an empty monitoreditemrequest, some don't.
 		// beter to just return
-		return nil
+		return nil, nil
 	}
 
 	toAdd := make([]*ua.MonitoredItemCreateRequest, 0)
 
-	for _, node := range nodes {
+	// Add handles
+	for i := range nodes {
 		handle := atomic.AddUint32(&s.monitor.nextClientHandle, 1)
-
-		s.handles[handle] = node
-		s.nodeLookup[node.String()] = &itemIDs{
-			handle: handle,
+		s.handles[handle] = nodes[i].NodeID
+		nodes[i].MonitoringParameters.ClientHandle = handle
```

Comment:
It's too bad we have to change the incoming request parameters, but I guess it can't be avoided.

Reply:
I left all the other incoming methods there and only added a new that accepts the new type Request. So this should not break any usage of this package as far as I can see.

Reply:
Ah, not the methods themselves. Rather the `nodes` param in `AddMonitorItems(nodes ...Request)`. It is a little unfortunate that the `nodes` param _changes_ as a result of calling the method. :shrug: